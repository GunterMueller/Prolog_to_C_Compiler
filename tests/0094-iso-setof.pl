legs(A, 6) :- insect(A).
legs(A, 4) :- animal(A).
legs(A, 8) :- spider(A).

insect(bee).
insect(ant).
animal(horse).
animal(cat).
animal(dog).
spider(tarantula).

main :-
	findall(N/B, setof(A, legs(A, N), B), L1),
	write(L1), nl,
	findall(S, setof(N-L, bagof(A, legs(A, N), L), S), L2),
	write(L2), nl,
	findall(S, setof(X, (X = 2; X = 2), S), L3),
	write(L3), nl,
	findall(S, setof(X, (X = Y; X = Z), S), L4),
	write(L4), nl,
	findall(S, setof(X, member(X,[V,U,f(U),f(V)]), S), L5),
	write(L5), nl.
