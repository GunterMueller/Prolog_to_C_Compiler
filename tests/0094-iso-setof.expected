[4/[cat,dog,horse],6/[ant,bee],8/[tarantula]]
[[4-[dog,cat,horse],6-[ant,bee],8-[tarantula]]]
[[2]]
[[_1593,_1594]]
[[_1949,_1950,f(_1949),f(_1950)]]
